cleverhans/attacks_tf.py

```diff
@@ -417,3 +417,245 @@ def jacobian_augmentation(sess, x, X_sub_prev, Y_sub, grads, lmbda,
 
     # Return augmented training data (needs to be labeled afterwards)
     return X_sub
+
+
+class CarliniWagnerL2:
```

Comment:
I think we already talked about that, but I'm wondering how we should organize the code here. This is the first attack that has an underlying class object. My question is do we want to eventually merge this `CarliniWagnerL2` class in `attacks_tf.py` w/ the wrapper class `CarliniWagnerL2` in `attacks.py`.

The reason this might make sense is that once we drop support for Theano in a few months, the wrapper classes in `attacks.py` will be pretty bare bones, so putting the code all in one place might make things easier to maintain (we could also apply the same logic to the other attacks).

Not sure what is the best way to go, @goodfeli do you have any thoughts?